CEXT-1539: Updating Custom Resolvers (#103)
AdobeDocs/graphql-mesh-gateway: src/pages/gateway/extending-unified-schema.md

					let max = 0;

					try {
						max = root.price_range.maximum_price.final_price.value;
					} catch (e) {
						// ignore
					}

					return context.DiscountsAPI.Query.discounts({
							root,
							args,
							context,
							info,
							selectionSet: '{ name discount }',
						})
						.then(response => {
							let discount = 0;
							const discountConfig = response.find(discount => discount.name === root.name);

							if (discountConfig) {
								discount = discountConfig.discount;
							}

							return max * ((100 - discount) / 100);
						})
						.catch(() => {
							return null;
						});
				},
			},
		},
	},
};
```

This `javascript` file creates a `special_price` field that uses the `maximum_price` for a product and then applies the discount listed for that product in the [`discountsapi.json` file](https://raw.githubusercontent.com/AdobeDocs/graphql-mesh-gateway/main/src/pages/_examples/discounts-api.json). The following arguments define how the `javascript` interacts with your mesh:

- `special_price` - the name of the new field we are creating

- `selectionSet` - the information selected from the source that you want to mutate

> - `root`, `context`, and `info` - mandatory parameters that are forwarded from the resolvers calling the method

Reviewer: ```suggestion
- `root`, `context`, `args`, and `info` - are parameters that are provided to the new resolver function
```

> - `args` - arguments to pass to the mutation or query

Reviewer: ```suggestion
```

```js
context.DiscountsAPI.Query.discounts({
    root,
    args,
    context,
    info,
    selectionSet: '{ name discount }',
})
```
calls the `discounts` query on the `DiscountsAPI` source through the `context` object. This results in a promise which resolves to the `discounts` data.
